worker/wallet.js

```diff
@@ -142,16 +142,25 @@ async function checkInvoice ({ data: { hash }, boss, models, lnd }) {
   }
 
   if (inv.is_held) {
-    // this is basically confirm_invoice without setting confirmed_at
+    // First query makes sure that after payment, HODL invoices are settled
+    // within 60 seconds or they will be canceled to minimize risk of
+    // force closures or wallets banning us.
+    // Second query is basically confirm_invoice without setting confirmed_at
     // and without setting the user balance
     // those will be set when the invoice is settled by user action
-    return await serialize(models, models.invoice.update({
-      where: { hash },
-      data: {
-        msatsReceived: Number(inv.received_mtokens),
-        isHeld: true
-      }
-    }))
+    const expiresAt = new Date(Math.min(dbInv.expiresAt, datePivot(new Date(), { seconds: 60 })))
+    return await serialize(models,
+      models.$queryRaw`
+      INSERT INTO pgboss.job (name, data, retrylimit, retrybackoff, startafter)
+      VALUES ('finalizeHodlInvoice', jsonb_build_object('hash', ${hash}), 21, true, ${expiresAt})`,
```

Comment:
This creates another job instead of updating the existing job but I think it's safer this way. Might be too paranoid though.